src/iamserver/iamserver.cpp

```diff
@@ -585,6 +586,47 @@ grpc::Status IAMServer::FinishProvisioning(
     return grpc::Status::OK;
 }
 
+grpc::Status IAMServer::Deprovision(grpc::ServerContext* context, const iamanager::v4::DeprovisionRequest* request,
```

Comment:
looks like the code for Deprovision and StartProvisioning is mostly the same, is it worth combining into one?